File: hw2.0/calcs/Android CALC MODERNED/Тут сам калькултор/кальк/calculator/app/src/main/java/com/example/calculator/Logic.java
package com.example.calculator;

import java.math.BigDecimal;

public class Logic
{
    public double summ(double a, double b) {
        a += b;
        if(!this.isNice(a)) return 1999999999;
        return this.checkLenght(a);
    }
    public double minus(double a, double b) {
        a -= b;
        if(!this.isNice(a)) return 1999999999;
        return this.checkLenght(a);
    }
    public boolean isNice(double a) {
        Integer c = (int) a;
        System.out.println(c + " ccccccccccccccc");
        if (c > 999999999) {
            return false;
        }
        return true;
    }
    public double divide(double a, double b) {
        if (a == 0 || b == 0) {
            return 0;
        }
        a /= b;
        if(!this.isNice(a)) return 1999999999;
        return this.checkLenght(a);
    }
    public double multiply(double a, double b) {
        a *= b;
        if(!this.isNice(a)) return 1999999999;
        return this.checkLenght(a);
    }
    public String cutDisplay(String a) {
        if (a == null) return "false";
        String Stringa = "";
        int i = 0;
        if (a.length() > 9)
        {
            for (char q : a.toCharArray())
            {
                i++;
                if (i > 9) break;
                Stringa += q;

            }
        }
        else
        {
            return a;
        }
        return Stringa;
    }
    public double checkLenght(Double a) {
        return Double.parseDouble(this.cutDisplay((new BigDecimal(a)).toString()));
    }
}

File: hw2.0/calcs/JS CALC MODERNED/Модернизированый/js/logic.js
function summ(a, b) {
	a = ((a * 1000000000) + (b * 1000000000))/1000000000;
	return a;
}
function minus(a, b) {
	a = ((a * 1000000000) - (b * 1000000000))/1000000000;
	return a;
}	
function divide(a, b) {
	a = ((a * 1000000000) / (b * 1000000000));
	return a; 
}
function multiply(a, b) {
	a = ((a * 1000000) * (b * 1000))/1000000000;
	return a;
}

let logic = {
	summ,
	minus,
	divide,
	multiply,
};

//пофиксил не корректную работу вычислениR джава скрипта этими странными цыфрами выше, не пугайтесь)